esphome/components/toshiba/toshiba.cpp

```diff
@@ -97,6 +99,300 @@ const std::vector<uint8_t> RAC_PT1411HWRU_TEMPERATURE_F{0x10, 0x30, 0x00, 0x20,
+//   Byte 0-1: Header (always 0xC23D)
+//   Byte 2-3: Fan speed encoding (varies by mode, see fan tables below)
+//   Byte 4:   Temperature + mode encoding
+//   Byte 5:   Bitwise complement of temperature byte
+//
+// Second Packet (rc_code_2): [D5] [FAN2] [00] [SUF1] [SUF2] [SUF3]
+//   Byte 0:   Header (always 0xD5)
+//   Byte 1:   Fan speed secondary encoding
+//   Byte 2:   Always 0x00
+//   Byte 3-5: Fixed suffix pattern (depends on fan speed and mode)
+//
+// TEMPERATURE ENCODING:
+// Base temp codes: 18°C=0x10, 19°C=0x30, 20°C=0x20, 21°C=0x60, 22°C=0x70,
+//                  23°C=0x50, 24°C=0x40, 25°C=0xC0, 26°C=0xD0, 27°C=0x90,
+//                  28°C=0x80, 29°C=0xA0, 30°C=0xB0
+// Mode offsets added to base temp:
+//   COOL: No offset
+//   HEAT: +0x0C (e.g., 24°C heat = 0x40 | 0x0C = 0x4C)
+//   AUTO: +0x08 (e.g., 24°C auto = 0x40 | 0x08 = 0x48)
+//   DRY:  +0x24 (e.g., 24°C dry = 0x40 | 0x24 = 0x64)
+//
+// FAN SPEED ENCODING (First packet bytes 2-3):
+//   AUTO: 0xBF40, QUIET: 0xFF00, LOW: 0x9F60, MEDIUM: 0x5FA0, HIGH: 0x3FC0
+//   Special cases: AUTO/DRY modes use 0x1FE0 instead
+//
+// SINGLE-PACKET COMMANDS:
+// Power Off: 0xC23D7B84E01F (6 bytes, no second packet)
+// Swing Toggle: 0xC23D6B94E01F (6 bytes, no second packet)
+//
+// MODE DETECTION (from first packet):
+// - Check bytes 2-3: if 0x7B84 → OFF mode
+// - Check bytes 2-3: if 0x1FE0 → AUTO/DRY/low-temp-COOL (distinguish by temp code)
+// - Otherwise: COOL/HEAT/FAN_ONLY (distinguish by temp code and byte 5)
+
+/**
+ * Get fan speed encoding for RAS-2819T first packet (rc_code_1, bytes 2-3)
+ */
+uint16_t get_ras_2819t_fan_code(climate::ClimateFanMode fan_mode) {
+  switch (fan_mode) {
+    case climate::CLIMATE_FAN_QUIET:
+      return RAS_2819T_FAN_QUIET;
+    case climate::CLIMATE_FAN_LOW:
+      return RAS_2819T_FAN_LOW;
+    case climate::CLIMATE_FAN_MEDIUM:
+      return RAS_2819T_FAN_MEDIUM;
+    case climate::CLIMATE_FAN_HIGH:
+      return RAS_2819T_FAN_HIGH;
+    case climate::CLIMATE_FAN_AUTO:
+    default:
+      return RAS_2819T_FAN_AUTO;
+  }
+}
+
+/**
+ * Get fan speed encoding for RAS-2819T rc_code_2 packet (second packet)
+ */
+struct Ras2819tSecondPacketCodes {
+  uint8_t fan_byte;
+  Ras2819tPacketSuffix suffix;
+};
+
+Ras2819tSecondPacketCodes get_ras_2819t_second_packet_codes(climate::ClimateFanMode fan_mode) {
+  switch (fan_mode) {
+    case climate::CLIMATE_FAN_QUIET:
+      return {RAS_2819T_FAN2_QUIET, RAS_2819T_SUFFIX_QUIET};
+    case climate::CLIMATE_FAN_LOW:
+      return {RAS_2819T_FAN2_LOW, RAS_2819T_SUFFIX_LOW};
+    case climate::CLIMATE_FAN_MEDIUM:
+      return {RAS_2819T_FAN2_MEDIUM, RAS_2819T_SUFFIX_MEDIUM};
+    case climate::CLIMATE_FAN_HIGH:
+      return {RAS_2819T_FAN2_HIGH, RAS_2819T_SUFFIX_HIGH};
+    case climate::CLIMATE_FAN_AUTO:
+    default:
+      return {RAS_2819T_FAN2_AUTO, RAS_2819T_SUFFIX_AUTO};
+  }
+}
+
+/**
+ * Get temperature code for RAS-2819T protocol
+ */
+uint8_t get_ras_2819t_temp_code(float temperature) {
```

Comment:
Since it's not operating on anything in the class:
```suggestion
static uint8_t get_ras_2819t_temp_code(float temperature) {
```